File: algorithms/search/linear_search/linear_search.java
class linear_search {
    public static int search(int arr[], int key ) {
        for( int i=0; i < arr.length; i++) {
            if(arr[i]==key) {
                return i;
            }
        }
        return -1;
    }

    public static void result(int idx) {
        if (idx == -1) {
            System.out.print("Element not found \n");
        } else {
            System.out.print("Element found at index " + idx + " \n");
        }
    }

    public static void main(String[] args) {
        int arr[] = {0, 42, 69, 17, -5, 2, 9, 1};
        result(search(arr, 3));
        result(search(arr, 2));
    }
}

File: algorithms/search/linear_search/linear_search.go
package main

import "fmt"

func search(datalist []int, key int) int {
	for i := 0; i < len(datalist); i++ {
		if datalist[i] == key {
			return i
		}
	}
	return -1
}

func result(idx int) {
	if idx == -1 {
		fmt.Println("Element not found")
	} else {
		fmt.Println("Element found at index", idx)
	}
}

func main() {
	arr := []int{0, 42, 69, 17, -5, 2, 9, 1}
	result(search(arr, 3))
	result(search(arr, 2))
}
